### PR TITLE
BWT-292: optimize performance of advisory dashboard on staff portal

### PR DESCRIPTION
**Pull request by mayurmarakana89**
### Jira Ticket:
BWT-292

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BWT-292

### Description:
Optimize performance of advisory dashboard on staff portal
- Load data once
- Fix data fetching due to react re-rendering

### CODE REVIEW

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 1]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
-              if (isMounted.current) {
-                setPublishedAdvisories([...publishedAdvisories]);
-              }
-            });
-        }
-      }
-    };
+    ])
+    .then((response) => {
+      getCurrentPublishedAdvisories(cmsData, setCmsData);
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:140]: Convert to async/await, not `.then()`

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 2]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
+                ad.advisoryNumber,
+              ];
+            });
+            if (isMounted.current) {
+              setPublishedAdvisories([...publishedAdvisories]);
+            }
+          })
+          .catch(() => {
+            setHasErrors(true);
+          });
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:120]: You're mixing both new ES6 await/async syntax and the old way of promise chaining `.then()`.  I would change it to the new format entirely.

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 3]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
+    fetchPublicAdvisory(selectedParkId);
+  }, [selectedParkId]);
+
+  const getCurrentPublishedAdvisories = async (cmsData, setCmsData) => {
+    const advisoryStatuses = await getAdvisoryStatuses(cmsData, setCmsData);
+    if (advisoryStatuses) {
+      const publishedStatus = advisoryStatuses.filter(
+        (as) => as.code === "PUB"
+      );
+      if (publishedStatus && publishedStatus[0]) {
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:100]: ```if (publishedStatus?.length > 0)``` is cleaner.

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 4]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
+      const publicAdvisories = response[1].data;
 
-    const regionParksCount = managementAreas.reduce((region, item) => {
-      region[item.region.id] =
-        (region[item.region.id] || 0) + item.protectedAreas.length;
-      return region;
-    }, {});
+      const regionParksCount = managementAreas.reduce((region, item) => {
+        region[item.region.id] =
+          (region[item.region.id] || 0) + item.protectedAreas.length;
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:147]: Can be one line.

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 5]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
-          region.count = regionParksCount[region.id];
-          regionsWithParkCount = [...regionsWithParkCount, region];
-        });
-        publicAdvisory.regions = regionsWithParkCount;
+      const updatedPublicAdvisories = publicAdvisories.map((publicAdvisory) => {
+        publicAdvisory.expired = publicAdvisory.expiryDate < today ? "Y" : "N";
+        publicAdvisory.associatedParks =
+          publicAdvisory.protectedAreas
+            .map((p) => p.protectedAreaName)
+            .join(", ") +
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:156]: Either One line/shorten the height

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js [hunk 6]
```diff
@@ -67,99 +71,113 @@ export default function AdvisoryDashboard({
...
+          publicAdvisory.regions.map((r) => r.regionName).join(", ");
+  
+        let regionsWithParkCount = [];
+        if (publicAdvisory.regions.length > 0) {
```
Comment (marklise) [inline src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:160]: Unsafe.  => ```if (publicAdvisory?.regions?.length > 0)```

**Review by marklise — CHANGES_REQUESTED** [review]

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 3 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:100]: done!

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 2 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:120]: done!

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 1 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:140]: done!

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 5 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:156]: Single line goes beyond 80 characters in one line hence it was written like that.

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 4 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:147]: Single line goes beyond 80 characters in one line hence it was written like that.

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 6 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js:160]: done!

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 4 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js]: done!

File: src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js (hunk 5 above)
Comment (mayurmarakana89) [reply src/admin/src/components/page/advisoryDashboard/AdvisoryDashboard.js]: done!

**Review by marklise — APPROVED** [review]

**Review by cameronpettit — APPROVED** [review]

**Pull request merged**